src/ensembl/production/metadata/grpc/adaptors/genome.py

```diff
@@ -647,3 +650,39 @@ def fetch_organisms_group_counts(self, species_taxonomy_id=None, release_version
 		with self.metadata_db.session_scope() as session:
 			# TODO check if we should return a dictionary instead
 			return session.execute(query).all()
+
+	def fetch_related_assemblies_count(self, species_taxonomy_id=None, release_version=None):
+		o_species = aliased(Organism)
+		o = aliased(Organism)
+		if not release_version:
+			# Get latest released organisms
+			query = db.select(
+				o_species.species_taxonomy_id,
+				o_species.ensembl_name,
+				o_species.common_name,
+				o_species.scientific_name,
+				db.func.count().label('count')
+			)
+
+			query = query.join(o, o_species.species_taxonomy_id == o.species_taxonomy_id)
+			query = query.join(Genome, o.organism_id == Genome.organism_id)
+			query = query.join(Assembly, Genome.assembly_id == Assembly.assembly_id)
+			query = query.filter(o_species.species_taxonomy_id == species_taxonomy_id)
+
+			query = query.group_by(
+				o_species.species_taxonomy_id,
+				o_species.ensembl_name,
+				o_species.common_name,
+				o_species.scientific_name,
+			)
+		else:
+			# change group to release_version_state and related genomes
+			raise NotImplementedError('Not implemented yet')
+			pass
+
+		# print(f"query ---> {query}")
+		with self.metadata_db.session_scope() as session:
+			results = session.execute(query).all()
```

Comment:
Was more thinking of something like: 

`session.execute(query).count()` 

will return all you need automatically, no need to reinvent the weel, so you don't do the `group` in the query :-)